service/src/main/java/com/epam/esm/mapper/GiftCertificateDtoMapper.java

```diff
@@ -0,0 +1,46 @@
+package com.epam.esm.mapper;
+
+
+import com.epam.esm.dto.GiftCertificateDto;
+import com.epam.esm.entities.GiftCertificate;
+import com.epam.esm.entities.Tag;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@Component
+public class GiftCertificateDtoMapper {
+
+    public GiftCertificateDto map(GiftCertificate giftCertificate) {
+        List <String> listOfTagsName = new ArrayList<>();
+        GiftCertificateDto giftCertificateDto = new GiftCertificateDto(
+                giftCertificate.getCertificateId(), giftCertificate.getName(),
+                giftCertificate.getDescription(), giftCertificate.getPrice(), giftCertificate.getDuration(),
+               giftCertificate.getCreateDate(),giftCertificate.getLastUpdateDate());
+
+        for (Tag tag : giftCertificate.getTags()) {
```

Comment:
it could be replaced with java stream api

Reply:
replaced with java stream api